source/common/common/token_bucket_impl.cc

```diff
@@ -40,8 +47,14 @@ std::chrono::milliseconds TokenBucketImpl::nextTokenAvailable() {
 
 void TokenBucketImpl::reset(uint64_t num_tokens) {
   ASSERT(num_tokens <= max_tokens_);
+  absl::WriterMutexLock lock(&mutex_);
+  // Don't reset if reset before and multiple resets aren't allowed.
+  if (reset_once_.has_value() && reset_once_.value()) {
```

Comment:
I see what you're doing here and why it's necessary for a thread-safe token bucket, but this behavior needs to be explained in comments in the header file. Ideally with an example.

Reply:
sure. will add explanation.